src/lexer.coffee

```diff
@@ -411,14 +411,20 @@ exports.Lexer = class Lexer
       value = @chunk.charAt 0
     tag  = value
     [..., prev] = @tokens
-    if value is '=' and prev
-      if prev.variable and prev[1] in JS_FORBIDDEN
-        prev = prev.origin if prev.origin
-        @error "reserved word '#{prev[1]}' can't be assigned", prev[2]
-      if prev[1] in ['||', '&&']
+
+    if prev and value in ['=', COMPOUND_ASSIGN...]
+      skipToken = false
+      if value is '=' and prev[1] in ['||', '&&'] and not prev.spaced
         prev[0] = 'COMPOUND_ASSIGN'
         prev[1] += '='
-        return value.length
+        prev = @tokens[@tokens.length - 2]
+        skipToken = true
+      if prev and prev.variable
+        origin = prev.origin ? prev
+        if message = isUnassignable prev[1], origin[1]
```

Comment:
I find this pattern of doing an assignment in a condition really unreadable and error-prone.